### ISSUE TITLE
Is the public API working already?

### ISSUE THREAD

**Issue by cryptoqween** [etherdelta/public_api#1]
Hi there, I was looking to get some data from you guys. Is the API working already?
Cheers

**Comment by zackcoburn** [comment]
Yes. Since it sounds like you are looking for general exchange data, I will point you here:

https://api.etherdelta.com/returnTicker

This looks just like Poloniex's returnTicker endpoint. It's a list of every token that has traded on the exchange along with its last traded price, 24 hour volume, and 24 hour percent change.

We also have a JS API (https://github.com/etherdelta/etherdelta.github.io/blob/master/api.js) that allows you to do things like check balances and trade (through the smart contract) and read the trade ticker. This API is currently undocumented, but there's a working example (https://github.com/etherdelta/etherdelta.github.io/blob/master/market_maker.js) that can be used to specify a basic market making strategy.

**Comment by cryptoqween** [comment]
Hi there, thanks for the quick reply.
I am actually looking for transaction data - preferably historical.

**Comment by cryptoqween** [comment]
I am looking to integrate to https://www.cryptocompare.com/

**Comment by zackcoburn** [comment]
It's possible to get via the Ethereum API or via API.js, but that's not the most convenient method. The plan is to add trade history to the api.etherdelta.com service in the future.

**Comment by cryptoqween** [comment]
Hi there, thanks for your help. 
We had many requests from users to include you guys on our website so would be great to get trades data at some point. Do you think you can add it sometime in the near future? Might boost the traffic for your site as well.

**Comment by zackcoburn** [comment]
Absolutely.

**Comment by mandelmonkey** [comment]
Is it possible to use an api to create orders? 
for example we could make a mobile interface to etherdelta to let users trade from their mobile device

**Comment by tsrbtc** [comment]
Is there an API call that will return a list of the pairs currently available on Etherdelta?

**Comment by zackcoburn** [comment]
I think this would serve your needs (see config.tokens): 

https://raw.githubusercontent.com/etherdelta/etherdelta.github.io/master/config.js
